src/main/java/io/github/eocqrs/kafka/data/WithPartition.java

```diff
@@ -55,7 +64,7 @@ public WithPartition(
     final Message<K, X> msg
   ) {
     this.partition = prtn;
-    this.message = msg;
+    this.message = new Synced<>(new Sticky<>(msg));
```

Comment:
@l3r8yJ why we are doing Sync here?

Reply:
@h1alexbel Due to the fact that `Sticky` is not thread-safe by default